Sources/NIO/EventLoopFuture.swift

```diff
@@ -1325,3 +1325,77 @@ extension EventLoopFuture {
         return self
     }
 }
+
+// MARK: unwrap
+
+extension EventLoopFuture {
+    /// Unwrap an `EventLoopFuture` where its type parameter is an `Optional`.
+    ///
+    /// Unwraps a future returning a new `EventLoopFuture` with the same value as the resolved future when its value
+    /// is `Optional.some(...)`. otherwise the `Error` passed in the `or` parameter is thrown. For example:
+    /// ```
+    /// do {
+    ///     try promise.futureResult.unwrap(or: ErrorToThrow).wait()
+    /// } catch ErrorToThrow {
+    ///     ...
+    /// }
+    /// ```
+    ///
+    /// - parameters:
+    ///     - or: the `Error` that is thrown when then resolved future's value is `Optional.none`.
+    /// - returns: an new `EventLoopFuture` with new type parameter `T` and the same value as the resolved
+    ///     future.
+    /// - throws: the `Error` passed in the `or` parameter when the resolved future's value is `Optional.none`.
+    func unwrap<T>(or error: Error) -> EventLoopFuture<T> where Value == T? {
+        self.flatMapThrowing { value in
+            guard let value = value else {
+                throw error
+            }
+            return value
+        }
+    }
+
+    /// Unwrap an `EventLoopFuture` where its type parameter is an `Optional`.
+    ///
+    /// Unwraps a future returning a new `EventLoopFuture` with either: the value passed in the `or`
+    /// parameter when the future resolved with value Optional.none, or the same value otherwise. For example:
+    /// ```
+    /// promise.futureResult.unwrap(or: 42).wait()
+    /// ```
+    ///
+    /// - parameters:
+    ///     - or: the value of the returned `EventLoopFuture` when then resolved future's value is `Optional.some()`.
+    /// - returns: an new `EventLoopFuture` with new type parameter `T` and the value passed in the `or` parameter.
+    func unwrap<T>(or replace: T) -> EventLoopFuture<T> where Value == T? {
+        self.flatMapThrowing { value in
```

Comment:
This isn't `flatMapThrowing` as the closure doesn't throw, it's just `map`.

Reply:
Yeah. I did change it to `flatMap` at one stage and it through up loads of errors. I obvious forgot about resolving that, I'll look again.

Reply:
Note that it's `map`, not `flatMap`.